pkg/virt-controller/watch/vmi/vmi_test.go

```diff
@@ -742,80 +742,44 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 	})
 
-	Context("with network-status annotation", func() {
+	Context("network annotations generation", func() {
 		const (
-			defaultNetworkName = "default"
-			sriovNetworkName   = "network1"
-			netAttachDefName   = "default/nad1"
-			selectedPCIAddress = "0000:04:02.5"
+			key1   = "key1"
+			value1 = "value1"
+			key2   = "key2"
+			value2 = "value2"
 		)
 
-		It("should not patch network-info annotation when no SR-IOV/binding plugin networks exist", func() {
+		It("should patch the pod's annotations when network annotations generator returns custom annotations", func() {
 			vmi := newPendingVirtualMachine("testvmi")
 			vmi.Status.Phase = virtv1.Running
-			vmi = addDefaultNetwork(vmi, defaultNetworkName)
+
 			pod := newPodForVirtualMachine(vmi, k8sv1.PodRunning)
-			pod.Annotations[networkv1.NetworkStatusAnnot] = `
-			[
-			{
-			"name": "kindnet",
-			"interface": "eth0",
-			"ips": [
-			  "10.244.2.131"
-			],
-			"mac": "82:cf:7c:98:43:7e",
-			"default": true,
-			"dns": {}
-			}
-			]`
+
 			addVirtualMachine(vmi)
 			addPod(pod)
 			addActivePods(vmi, pod.UID, "")
 
+			controller.netAnnotationsGenerator = stubNetworkAnnotationsGenerator{
+				annotations: map[string]string{key1: value1, key2: value2},
+			}
 			controller.Execute()
-			expectPodAnnotations(pod, Not(HaveKey(downwardapi.NetworkInfoAnnot)))
+			expectPodAnnotations(pod, HaveKeyWithValue(key1, value1), HaveKeyWithValue(key2, value2))
 		})
 
-		It("should patch network-info annotation when SR-IOV networks exist", func() {
+		It("should not patch the pod's annotations when network annotations generator returns an empty map", func() {
```

Comment:
Can you set these annotations explicitly empty? It is now implicitly empty, which is hard to follow.

Reply:
Done.